### PR TITLE
Couchbase vector store support as destination and source connector

### PR DESCRIPTION
**Pull request by lokesh-couchbase**
Couchbase is a Key-Value based NoSQL Database. 

This PR intends to add support for couchbase as a source and destination connector to unstructured.io

### CODE REVIEW

**Comment by potter-potter** [comment]
Hey @lokesh-couchbase this looks super cool! 

For the tests, could you build them to run off of Docker containers (for both source and destination) instead of using the hosted service.

This will make sure the couchbase connectors are ready to be used in both hosted and open source couchbase.

It will also help us in that we don't have to subscribe to couchbase just to maintain a connector. (the source/destination tests run everytime CI/CD is run.)

You can see some good examples in the elasticsearch, chroma or kafka tests.

https://github.com/Unstructured-IO/unstructured/blob/3f581e6b7d4ce4b45e6fdb58520047441c21d96c/test_unstructured_ingest/dest/elasticsearch.sh#L36

https://github.com/Unstructured-IO/unstructured/blob/main/scripts/elasticsearch-test-helpers/destination_connector/create-elasticsearch-instance.sh

**Comment by potter-potter** [comment]
@lokesh-couchbase I'm checking this out now. Thanks for the docker container...

File: test_unstructured_ingest/dest/couchbase.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+wait
+
+
+PYTHONPATH=. ./unstructured/ingest/main.py \
+  local \
+  --num-processes "$max_processes" \
+  --output-dir "$OUTPUT_DIR" \
+  --strategy fast \
+  --verbose \
+  --input-path example-docs/book-war-and-peace-1225p.txt \
```
Comment (potter-potter) [inline test_unstructured_ingest/dest/couchbase.sh:73]: lets switch this to the 1 page book
book-war-and-peace-1p.txt

File: scripts/couchbase-test-helpers/ingest_setup_cluster.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+                            },
+                        },
+                    }
+                },
+            },
+            "store": {"indexType": "scorch", "segmentVersion": 16},
+        },
+        "sourceParams": {},
+    }
+    scope_index_manager = scope.search_indexes()
```
Comment (potter-potter) [inline scripts/couchbase-test-helpers/ingest_setup_cluster.py:209]: Does there need to be a sleep or something between this line and the next line?
I keep getting an error if I try to run this script.
But if I put a breakpoint and wait a while. And then continue it works fine...

File: test_unstructured_ingest/dest/couchbase.sh [hunk 3]
```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
```
Comment (potter-potter) [inline test_unstructured_ingest/dest/couchbase.sh:1]: please make sure this file's permissions are runnable. `chmod 755`

File: test_unstructured_ingest/dest/couchbase.sh (hunk 1 above)
Comment (thelokeshgoel00) [reply test_unstructured_ingest/dest/couchbase.sh]: updated

File: scripts/couchbase-test-helpers/ingest_setup_cluster.py (hunk 2 above)
Comment (lokesh-couchbase) [reply scripts/couchbase-test-helpers/ingest_setup_cluster.py]: I also found this issue. I have added appropriate health checks wherever I saw some issues related to this. Please let me know if you still encounter anything like this again

File: test_unstructured_ingest/dest/couchbase.sh (hunk 3 above)
Comment (potter-potter) [reply test_unstructured_ingest/dest/couchbase.sh:1]: Before you check it in, check the permissions locally to see if it is runnable. I believe that works. The next person that downloads the branch should just be able to run it with `./test_unstructured_ingest/dest/couchbase.sh`

If not, I can always do it later.

Comment (lokesh-couchbase) [reply test_unstructured_ingest/dest/couchbase.sh:1]: updated

**Comment by thelokeshgoel00** [comment]
Hey @potter-potter, @rbiseck3 . I have started working on v2 destination connector. Currently I am facing some issues while running the destination connector.

The error is about not having embed-huggingface but it's present and I have retried to install it as well. I have also tried to reinstall all packages but still got the same error. Do you know what may be the error or maybe I am missing something?

Logs:

```2024-08-01 19:06:44,825 MainProcess INFO     Created index with configs: {"input_path": "example-docs/book-war-and-peace-1p.txt", "recursive": false, "file_glob": null}, connection configs: {"access_config": {}}
2024-08-01 19:06:44,825 MainProcess INFO     Created download with configs: {"download_dir": null}, connection configs: {"access_config": {}}
2024-08-01 19:06:44,826 MainProcess INFO     Created partition with configs: {"strategy": "fast", "ocr_languages": null, "encoding": null, "additional_partition_args": null, "skip_infer_table_types": null, "fields_include": ["element_id", "text", "type", "metadata", "embeddings"], "flatten_metadata": false, "metadata_exclude": [], "metadata_include": [], "partition_endpoint": "https://api.unstructured.io/general/v0/general", "partition_by_api": false, "api_key": null, "hi_res_model_name": null}
2024-08-01 19:06:44,826 MainProcess INFO     Created chunk with configs: {"chunking_strategy": "by_title", "chunking_endpoint": "https://api.unstructured.io/general/v0/general", "chunk_by_api": false, "chunk_api_key": null, "chunk_combine_text_under_n_chars": null, "chunk_include_orig_elements": true, "chunk_max_characters": 1500, "chunk_multipage_sections": true, "chunk_new_after_n_chars": null, "chunk_overlap": 0, "chunk_overlap_all": false}
2024-08-01 19:06:44,826 MainProcess INFO     Created embed with configs: {"embedding_provider": "langchain-huggingface", "embedding_api_key": null, "embedding_model_name": null, "embedding_aws_access_key_id": null, "embedding_aws_secret_access_key": null, "embedding_aws_region": "us-west-2"}
2024-08-01 19:06:44,826 MainProcess ERROR    failed to run destination command couchbase: Error in connecting to the embedding model provider: Following dependencies are missing: langchain_huggingface. Please install them using `pip install "unstructured[embed-huggingface]"`.
Traceback (most recent call last):
  File "/Users/lokeshgoel/Documents/couchbase/unstructured/unstructured/ingest/error.py", line 19, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/Users/lokeshgoel/Documents/couchbase/unstructured/unstructured/utils.py", line 248, in wrapper
    run_check()
  File "/Users/lokeshgoel/Documents/couchbase/unstructured/unstructured/utils.py", line 237, in run_check
    raise ImportError(
ImportError: Following dependencies are missing: langchain_huggingface. Please install them using `pip install "unstructured[embed-huggingface]"`.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/lokeshgoel/Documents/couchbase/unstructured/unstructured/ingest/v2/cli/base/dest.py", line 29, in cmd
    pipeline = self.get_pipline(
               ^^^^^^^^^^^^^^^^^
  File "/Users/lokeshgoel/Documents/couchbase/unstructured/unstructured/ingest/v2/cli/base/cmd.py", line 94, in get_pipline
    return Pipeline(**pipeline_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 4, in __init__
  File "/Users/lokeshgoel/Documents/couchbase/unstructured/unstructured/ingest/v2/pipeline/pipeline.py", line 76, in __post_init__
    embedder.config.get_embedder().initialize()
  File "/Users/lokeshgoel/Documents/couchbase/unstructured/unstructured/embed/huggingface.py", line 45, in initialize
    _ = self.client
        ^^^^^^^^^^^
  File "/Users/lokeshgoel/Documents/couchbase/unstructured/unstructured/embed/huggingface.py", line 34, in client
    self._client = self.create_client()
                   ^^^^^^^^^^^^^^^^^^^^
  File "/Users/lokeshgoel/Documents/couchbase/unstructured/unstructured/ingest/error.py", line 22, in wrapper
    raise cls(cls.error_string.format(str(error))) from error
unstructured.ingest.error.EmbeddingEncoderConnectionError: Error in connecting to the embedding model provider: Following dependencies are missing: langchain_huggingface. Please install them using `pip install "unstructured[embed-huggingface]"`.
Error: Error in connecting to the embedding model provider: Following dependencies are missing: langchain_huggingface. Please install them using `pip install "unstructured[embed-huggingface]"`.
```

**Comment by lokesh-couchbase** [comment]
Merged on unstructured-ingest

**Pull request closed without merging**